From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Tue, 6 Oct 2026 20:09:47 +0200
Subject: [PATCH] build: Mark timer test as flaky on x86

This has been a long-standing problem with floating-point precision.

See: https://gitlab.gnome.org/GNOME/glib/-/work_items/820
---
 glib/tests/meson.build | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/glib/tests/meson.build b/glib/tests/meson.build
index 9d051ca27e66..74e04418a6eb 100644
--- a/glib/tests/meson.build
+++ b/glib/tests/meson.build
@@ -186,7 +186,10 @@ glib_tests = {
   'thread-pool' : {},
   'thread-pool-slow' : {'suite' : ['slow']},
   'timeout' : {},
-  'timer' : {},
+  'timer' : {
+    # FIXME: https://gitlab.gnome.org/GNOME/glib/-/work_items/820
+    'can_fail': host_machine.cpu_family() == 'x86',
+  },
   'tree' : {},
   'types' : {},
   'utf8-performance' : {},
